### PR TITLE
chore: fixed the form text fields validation UI

### PR DESCRIPTION
**Pull request by mohammedDehairy**
chore: fixed the form text fields validation UI

### CODE REVIEW

File: Adyen/UI/Form/Items/Text/FormTextItemView.swift [hunk 1]
```diff
@@ -292,12 +292,13 @@ open class FormTextItemView<ItemType: FormTextItem>: FormValueItemView<String, F
 
     /// :nodoc:
     open func updateValidationStatus(forced: Bool = false) {
-        if item.isValid() {
+        let forceShowValidationStatus = forced || !(textField.text ?? "").isEmpty
```
Comment (kaphacius) [inline Adyen/UI/Form/Items/Text/FormTextItemView.swift:295]: i think this is still a bit hard to read. maybe
```swift
let hasText = textField.text.map(\.isEmpty) == false
let forceShowValidationStatus = forced || hasText
```

File: Adyen/Helpers/UIViewHelpers.swift [hunk 2]
```diff
@@ -54,6 +55,7 @@ extension AdyenScope where Base: UIView {
         guard base.isHidden != hidden else { return }
         
         base.isHidden = hidden
+        base.alpha = hidden ? 0.0 : 1.0
```
Comment (kaphacius) [inline Adyen/Helpers/UIViewHelpers.swift:58]: *`0` and `1`

File: Adyen/UI/Form/Items/Value/FormValueItemView.swift [hunk 3]
```diff
@@ -83,13 +83,18 @@ open class FormValueItemView<ValueType, Style, ItemType: FormValueItem<ValueType
...
-        separatorView.backgroundColor = item.style.separatorColor ?? UIColor.Adyen.componentSeparator
+        separatorView.backgroundColor = defaultSeparatorColor
         separatorView.isUserInteractionEnabled = false
         separatorView.translatesAutoresizingMaskIntoConstraints = false
         
         return separatorView
     }()
     
+    internal var defaultSeparatorColor: UIColor {
+        guard !isFirstResponder else { return tintColor }
```
Comment (kaphacius) [inline Adyen/UI/Form/Items/Value/FormValueItemView.swift:94]: maybe better to use `if {...} else {...}` here. or a ternary operator :)

**Review by kaphacius — APPROVED** [review]

**Review by kaphacius — APPROVED** [review]

File: Adyen/UI/Form/Items/Text/FormTextItemView.swift [hunk 4]
```diff
@@ -292,12 +292,14 @@ open class FormTextItemView<ItemType: FormTextItem>: FormValueItemView<String, F
 
     /// :nodoc:
     open func updateValidationStatus(forced: Bool = false) {
-        if item.isValid() {
+        let hasText = textField.text.map(\.isEmpty) == false
```
Comment (nauaros) [inline Adyen/UI/Form/Items/Text/FormTextItemView.swift:295]: What do you think about renaming this variable to something more descriptive like `isTextFieldEmpty` and rewrite the condition as `force || !isTextFieldEmpty`?

Or keep it like this and just name it `isTextFieldNotEmpty`

Comment (mohammedDehairy) [reply Adyen/UI/Form/Items/Text/FormTextItemView.swift]: 👍

**Review by kaphacius — APPROVED** [review]

**Review by nauaros — APPROVED** [review]

**Pull request merged**